internal/tekton/client.go

```diff
@@ -0,0 +1,81 @@
+package tekton
+
+import (
+	"errors"
+
+	"github.com/opendevstack/pipeline/pkg/logging"
+	tektonClient "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	v1beta1 "github.com/tektoncd/pipeline/pkg/client/clientset/versioned/typed/pipeline/v1beta1"
+	"k8s.io/client-go/rest"
+)
+
+// Client represents a Tekton client, wrapping
+// github.com/tektoncd/pipeline/pkg/client/clientset/versioned.Clientset
+type Client struct {
+	clientConfig *ClientConfig
+}
+
+// ClientConfig configures a Tekton client.
+type ClientConfig struct {
+	// Kubernetes namespace.
+	Namespace string
+	// Logger is the logger to send logging messages to.
+	Logger logging.LeveledLoggerInterface
+	// HTTP client used to download assets.
+	TektonClient *tektonClient.Clientset
+}
+
+type ClientInterface interface {
+	ClientPipelineInterface
+}
+
+// NewInCluserClient initializes a Tekton client from within a cluster.
+func NewInCluserClient(clientConfig *ClientConfig) (*Client, error) {
```

Comment:
I think we've got typo in the function name (should read `NewInClusterClient`)

Reply:
Fixed in 1103911.